tss-esapi-sys/build.rs

```diff
@@ -1,114 +1,487 @@
+                            PATH_ENV_VAR_NAME, invalid_value
+                        );
+                    }
+                    std::env::VarError::NotPresent => None,
+                },
+                |var| Some(Installation::ensure_valid_installation_path(var)),
+            )
+        }
+
+        /// Ensures that the installation path is valid.
+        ///
+        /// # Details
+        /// In order to be considered valid the following
+        /// requirements needs to be full filled:
+        /// 1. The directory must exist.
+        /// 2. Sub directories `include` and `lib` must exist.
+        /// 3. A `VERSION` file must be present in the directory and it needs to be
+        ///    be specifying a version that is greater then the minimum supported version.
+        ///
+        /// The function is also responsible for reporting the library search path the rust compiler
+        /// should use.
+        ///
+        /// # Arguments
+        /// env_var - The value of the environment variable that contains the installation path.
+        ///
+        /// # Returns
+        /// A tuple containing the validated installation path and the version associated with it.
+        fn ensure_valid_installation_path(env_var: String) -> (PathBuf, String) {
+            let install_path = PathBuf::from(env_var);
+            if !install_path.is_dir() {
+                panic!(
+                    "`{}` specifies a path `{}`, that does not exist",
+                    PATH_ENV_VAR_NAME,
+                    install_path.to_string_lossy()
+                );
+            }
+            if !install_path.join("include").is_dir() {
+                panic!(
+                    "`{}` specifies a path `{}`, that does not contain an `include` directory",
+                    PATH_ENV_VAR_NAME,
+                    install_path.to_string_lossy()
+                );
+            }
+            if !install_path.join("lib").is_dir() {
+                panic!(
+                    "`{}` specifies a path `{}`, that does not contain an `lib` directory",
+                    PATH_ENV_VAR_NAME,
+                    install_path.to_string_lossy()
+                );
+            }
+            let version_str = std::fs::read_to_string(install_path.join("VERSION"))
+                .expect("Failed to read VERSION file.");
+            let version = Version::parse(version_str.trim())
+                .expect("Failed to parse the content of the VERSION file.");
+            let min_version_req_str = format!(">={}", MINIMUM_VERSION);
+            let min_version_req = VersionReq::parse(&min_version_req_str)
+                .expect("Failed to parse minimum version requirement.");
+            if !min_version_req.matches(&version) {
+                panic!(
+                    "Minimum supported version is {}, found installed version {}",
+                    MINIMUM_VERSION, version_str
+                );
+            }
```

Comment:
Could you also add a bit in these expect strings to make it clear which version we mean? Something like `Failed to read tpm2-tss VERSION file`.

Reply:
Absolutely

Reply:
Looks great, thank you